com.unity.shadergraph/Editor/GraphDeltaRegistry/Generation/Interpreter.cs

```diff
@@ -109,7 +109,10 @@ internal static Block EvaluateGraphAndPopulateDescriptors(INodeReader rootNode,
             }
             else
             {
-                mainBodyFunctionBuilder.AddLine($"output.{outputVariables[0].ReferenceName} = SYNTAX_{rootNode.GetName()}_{rootNode.GetOutputPorts().First().GetName()};");
+                var port = rootNode.GetOutputPorts().First();
+                var outType = registry.GetTypeBuilder(port.GetRegistryKey()).GetShaderType((IFieldReader)port, container, registry);
+                string assignment = ConvertToFloat3(outType, $"SYNTAX_{rootNode.GetName()}_{rootNode.GetOutputPorts().First().GetName()}");
+                mainBodyFunctionBuilder.AddLine($"output.{outputVariables[0].ReferenceName} = {assignment};");
```

Comment:
This just injects additional HLSL if there is a type mismatch on the port used for preview output.

Reply:
Need to add a test for this.

Reply:
Test added for this and previous case.